src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/SubmissionInterceptor.java

```diff
@@ -32,18 +32,19 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
         final Map<String, String> pathVariables = (Map<String, String>) request
                 .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
         final var submissionId = pathVariables.get(CONFIRMATION_STATEMENT_ID_KEY);
+        final var transactionId = pathVariables.get(TRANSACTION_ID_KEY);
 
         var logMap = new HashMap<String, Object>();
         String reqId = request.getHeader(ERIC_REQUEST_ID_KEY);
 
         Optional<ConfirmationStatementSubmissionJson> submission = confirmationStatementService.getConfirmationStatement(submissionId);
         boolean isPresent = submission.isPresent();
-        if (isPresent) {
-            ApiLogger.infoContext(reqId, "Submission found for " + submissionId, logMap);
+
+        if (isPresent && submission.get().getLinks().get("self").contains(transactionId)) {
```

Comment:
why do we need to check the self link contains the transaction id?
And how the code currently is, if the transaction id isnt in the submission self link, it returns not found, which also doesnt seem right.

But I am confised where this requirement has came from